### ISSUE TITLE
ModalHandler in testpage not open the page

### ISSUE THREAD

**Issue by stefanoPrimo** [Microsoft/AL#4275]
I'm trying to handle a Modal Page in a test Codeunit.
The problem is when the modal page is trapped the TestPage Object is returned "Closed".

this is the code:
        ModalTestPage.Trap();
        CustomerCard.Compensation.Invoke();  //this call the modal page 

I have also added the ModalPageHandler:
[ModalPageHandler]
    procedure MyModalPageHandler(var a: TestPage "My Page")
    begin
    end;

Is someone facing the same issue?
Thanks

**Comment by kalberes** [comment]
Which versions are we talking about?

**Comment by stefanoPrimo** [comment]
Business Central W1 RTM
Al Language: 2.0.54943

**Comment by ashik-ivor** [comment]
[ModalPageHandler]
procedure MyModalPageHandler(var a : TestPage "My Page")
begin
a.setvalue();
end;

You can try without using trap function also. 

You can handle the page in the handler and assign values to it.

**Comment by stefanoPrimo** [comment]
Yes, I know that, but I would like to keep my Handlers mostly generic so I can use it for multiple purposes.
Thanks anyway

**Comment by esbenk** [comment]
This is by design. The Trap() functionality is only designed for Non-Modal pages.
